pkg/tfgen/generate_nodejs.go

```diff
@@ -675,8 +676,14 @@ func (g *nodeJSGenerator) emitOverlay(mod *module, overlay *overlayFile) (string
 	// Copy the file from the overlays directory to the destination.
 	dir := g.moduleDir(mod)
 	dst := filepath.Join(dir, overlay.name)
-	if err := copyFile(overlay.src, dst); err != nil {
-		return "", err
+	if overlay.Copy() {
+		if err := copyFile(overlay.src, dst); err != nil {
+			return "", err
+		}
+	} else {
+		if _, err := os.Stat(dst); err != nil {
+			return "", err
+		}
 	}
```

Comment:
my only mild feeback is that i don't like that this is effectivly identical to the code in generate_go, and it would be nice to share hte logic.